libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -203,6 +203,9 @@ void ContractLevelChecker::checkFunctionOverride(FunctionDefinition const& _func
 			stateMutabilityToString(_function.stateMutability()) +
 			"\"."
 		);
+	if (_function.visibility() == Declaration::Visibility::Private)
```

Comment:
This actually checks "private functions cannot override". Can you add a test for different combinations of publc / private in the base an derived?

Reply:
Done. All fail as they should :)

Reply:
But the error messages are wrong: https://github.com/ethereum/solidity/pull/8455/files#diff-302ae33fa6301f5b2401b6e8cd02484a

You need to test for `_super.visibility()`

Reply:
oh, true. Updated